### PR TITLE
fix(influx): add env vars to cli usage and normalize usage and flag/env var priority

### LINKED ISSUE #16048: Add environment variables to the influx --help command
in the 1.x world, the --help command would show the environment variables that could be set as well so that the command could be shorter. I know about `INFLUX_TOKEN` but are there others for the ORG or BUCKET as well?

It would be helpful to have them here:

```
➜  influxdb git:(master) ✗ ./bin/darwin/influx pkg
Error: required flag(s) "org-id" not set
Usage:
  influx pkg [flags]
  influx pkg [command]

Available Commands:
  export      Export existing resources as a package
  new         Create a reusable pkg to create resources in a declarative manner
  summary     Summarize the provided package
  validate    Validate the provided package

Flags:
  -c, --color               Enable color in output, defaults true (default true)
  -f, --file string         Path to package file
      --force-on-conflict   TTY input, if package will have destructive changes, proceed if set true. (default true)
  -h, --help                Help for the pkg command
  -o, --org-id string       The ID of the organization that owns the bucket
  -q, --quiet               disable output printing
      --table-borders       Enable table borders, defaults true (default true)

Global Flags:
      --host string    HTTP address of Influx (default "http://localhost:9999")
      --local          Run commands locally against the filesystem
      --skip-verify    SkipVerify controls whether a client verifies the server's certificate chain and host name.
  -t, --token string   API token to be used throughout client calls

Use "influx pkg [command] --help" for more information about a command.
```

### LINKED ISSUE #16246: cli flags should override environmental variables
We use env vars to supplement our CLI, e.g. allowing users to scope all requests to a specific org without having to specify it in every request.

If a user specifies a flag that conflicts with an env var, the flag should take precedence over the env var.

1) Investigate where this is currently happening in the CLI.
2) Change all instances where env var takes precedence over flags.

### PR DESCRIPTION
**Pull request by jsteenb2**
also rids the pkg of the  cobra tutorial init style and encapsulates things into funcs. One step closer to builder pattern for this cmd pkg.

Closes #16246
Closes #16048 

- [x] [CHANGELOG.md](https://github.com/influxdata/influxdb/blob/master/CHANGELOG.md) updated with a link to the PR (not the Issue)
- [x] [Well-formatted commit messages](https://www.conventionalcommits.org/en/v1.0.0-beta.3/)
- [x] Rebased/mergeable
- [x] Tests pass
- [x] Documentation updated or issue created (provide link to issue/pr) https://v2.docs.influxdata.com/v2.0/reference/cli/influx

### CODE REVIEW

**Comment by jsteenb2** [comment]
@sanderson this will make a bunch of changes to the CLI help outputs. The env vars will be exposed so folks will know what they are when they see the `-h` usage.

File: cmd/influx/authorization.go [hunk 1]
```diff
@@ -226,15 +218,15 @@ func authorizationCreateF(cmd *cobra.Command, args []string) error {
...
+		OrgID:       orgID,
 	}
 
 	if userName := authCreateFlags.user; userName != "" {
 		userSvc, err := newUserService()
 		if err != nil {
 			return err
 		}
-		user, err := userSvc.FindUser(ctx, platform.UserFilter{
+		user, err := userSvc.FindUser(context.Background(), platform.UserFilter{
```
Comment (kelwang) [inline cmd/influx/authorization.go]: i think we would like to hook up the context sometime in the future, that's why the ctx is defined on top

File: cmd/influx/authorization.go [hunk 2]
```diff
@@ -299,23 +287,13 @@ func authFindCmd() *cobra.Command {
...
-	if h := viper.GetString("ORG_ID"); h != "" {
-		authorizationFindFlags.orgID = h
-	}
 	cmd.Flags().StringVarP(&authorizationFindFlags.id, "id", "i", "", "The authorization ID")
 
 	return cmd
 }
 
-func newAuthorizationService(f Flags) (platform.AuthorizationService, error) {
+func newAuthorizationService() (platform.AuthorizationService, error) {
```
Comment (kelwang) [inline cmd/influx/authorization.go]: I am never a big fan of anything global. If passing the flags bothers you, you could compose a struct and make this function a method.

File: cmd/influx/main.go [hunk 3]
```diff
@@ -73,32 +75,11 @@ func out(w io.Writer) genericCLIOptfn {
...
-	}
-	return cmd
-}
-
-func seeHelp(c *cobra.Command, args []string) {
-	if c = fetchSubCommand(c, args); c == nil {
-		return //return here, since cobra already handles the error
-	}
-	c.Printf("See '%s -h' for help\n", c.CommandPath())
+var flags struct {
```
Comment (kelwang) [inline cmd/influx/main.go]: i don't really like such a global thing.

File: cmd/influx/authorization.go (hunk 1 above)
Comment (jsteenb2) [reply cmd/influx/authorization.go]: I wish that were the case, but the context that was removed above waws just `context.Background()` 🤷‍♂. It just happened to be instantiated well before its use case.

File: cmd/influx/main.go (hunk 3 above)
Comment (jsteenb2) [reply cmd/influx/main.go]: totally agree. I didn't remedy all the issues in this pkg on this go round. However, I agree, the global constructs in this are a major issue and should be fixed. I didn't do it here b/c it was already bigger body of work than I anticipated 😢

File: cmd/influx/authorization.go (hunk 2 above)
Comment (jsteenb2) [reply cmd/influx/authorization.go]: totally agree. I'd like to see all the commands use the builder pattern of sorts so we can dep inject. We'r enot quite there yet. That'll take some time.

**Pull request merged**